src/coerce.c

```diff
@@ -383,7 +383,7 @@ coerce_into_file(struct workspace *wk, struct coerce_into_files_ctx *ctx, obj va
 			}
 			break;
 		case mode_output:
-			if (!path_is_basename(get_cstr(wk, val))) {
+			if (!path_is_basename(get_cstr(wk, val)) && wk->lang_mode != language_extended) {
```

Comment:
To be honest, this is the part of the PR that I am most unsure about so I am pleased that are fine with it.

I might put a TODO here based on your comment about this being a workaround.

For any future readers, this change was necessary because the `i18n` module in upstream meson creates `custom_target`s without the limitation that is placed on the output of user created `custom_target`s that no output path may contain a valid file separator.

I do wonder how, over the longer term, as more and more modules are implemented in muon's extended language, how the project aims to manage in a sensible way any semantic divergences between the extended language and the standard language. I should probably actually update the documentation to reflect this newly changed behaviour.

Reply:
This ties into the meson proposal that a number of people want, whereby you should be able to specify output paths with a file separator ***as long as*** the output directory:
- is a pure subpath, no `../` in it
- doesn't exist in the source tree

i.e. you're not using it to produce spaghetti projects where any target can be unpredictably defined anywhere. You can only use it to emit files into a tool-defined layout beneath the current directory.

Oddly enough that's exactly what the gettext method is intending -- I designed that gettext approach including the fact that it ignores the usual constraints.